python/paddle/fleet/meta_optimizers/dgc_optimizer.py

```diff
@@ -0,0 +1,87 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+from paddle.fluid.optimizer import Momentum, DGCMomentumOptimizer
+from .meta_optimizer_base import MetaOptimizerBase
+import logging
+
+__all__ = ["DGCOptimizer"]
+
+
+class DGCOptimizer(MetaOptimizerBase):
+    def __init__(self, optimizer):
+        super(DGCOptimizer, self).__init__(optimizer)
+        self.inner_opt = optimizer
+        self.dgc_opt = None
+        # we do not allow meta optimizer to be inner optimizer currently
+        self.meta_optimizers_white_list = []
+
+    def _set_basic_info(self, loss, role_maker, user_defined_optimizer,
+                        user_defined_strategy):
+        super(DGCOptimizer, self)._set_basic_info(
+            loss, role_maker, user_defined_optimizer, user_defined_strategy)
+
+        opt = self.inner_opt
+        if not isinstance(opt, Momentum):
+            return
+
+        learning_rate = opt._learning_rate
+        parameter_list = opt._parameter_list
+        regularization = opt.regularization
+        grad_clip = opt._grad_clip
+        name = opt._name
+        momentum = opt._momentum
+        use_nesterov = opt._use_nesterov
+
+        rampup_begin_step = 0
+        rampup_step = 1
+        sparsity = [0.999]
+
+        num_trainers = self.role_maker.worker_num()
+
+        self.dgc_opt = DGCMomentumOptimizer(
+            learning_rate=learning_rate,
```

Comment:
why not directly assign `opt._learning_rate`?

Reply:
Done.